### DISCUSSION TITLE
aardvark doesn't reply to hosts added by --add-host

### DISCUSSION THREAD

**Discussion by grisu48** [containers/podman#22829] [Q&A] [answered]
I'm working on putting a certain application into a podman pod, which relies on DNS resolution of internal hosts but refuses to consult `/etc/hosts`. I was hoping that aardvark-dns would be the saviour but it appears to me, that hosts added by `--add-host` are not being resolved by aardvark-dns.

See the following reproducer, where I add a new host `mulefta` but the aardvark-dns server doesn't resolve it's IP address:

```bash
#!/bin/bash

function cleanup() {
        # container ignores SIGTERM
        podman container kill toolbox3

        podman pod stop pod3
        podman container rm toolbox3
        podman pod rm pod3
        podman network rm podman3
}
trap cleanup EXIT

podman network create --subnet 10.90.1.0/24 podman3
podman pod create --network=podman3 --name=pod3 --add-host mulefa:10.90.1.10
podman create --pod pod3 --name toolbox3 registry.opensuse.org/opensuse/toolbox sleep infinity
podman pod start pod3
# host is not installed by default in the toolbox container
podman exec toolbox3 zypper -n in bind-utils >/dev/null

# Always returns "Host mulefa not found: 3(NXDOMAIN)"
podman exec toolbox3 host mulefa
podman exec toolbox3 host mulefa 10.90.1.1
```

Here the result for the last two requests is always "Host mulefa not found"

```
Host mulefa not found: 3(NXDOMAIN)
Using domain server:
Name: 10.90.1.1
Address: 10.90.1.1#53
Aliases: 

Host mulefa not found: 3(NXDOMAIN)
```

Am I doing something wrong here?

**Comment by Luap99** [comment, marked as answer]
--add-hosts only controls the writing of /etc/hosts so this has nothing to do with dns/aardvark-dns

If you want specific dns names to aardvark-dns then you must use `--network-alias mulefa` which add the alias to all networks the container is part of  or `--network podman3:alias=mulefa` to only add the name for a specific network

**Reply by grisu48** [reply]
Thank you, I will try to see if that resolved my specific problem.

Our of curiosity - I was wrongly assuming that aardvark-dns also resolves entries from it the `/etc/hosts` from its pod. Is there a reason why it doesn't? :thinking:

**Reply by Luap99** [reply]
Because hosts file and dns are different things. Sure people may use it for the same thing but they are not.
 --add-host adds a host entry for this single one pod/container, a network alias is added for the network which means all other containers on the same network can resolve it by this name and is automatically resolved to the correct container ip.

Aardvark-dns cannot read anything from pods/containers, it gets a list of networks with ips and names.

**Reply by grisu48** [reply]
Yes that makes sense. I now understand that they are two different contexts that often overlap but not necessarily do.

Thanks for the explanation.

**Reply by grisu48** [reply]
For my particular problem: adding a `--network=alias=mulefa` within the `podman pod create` command helped. Thank you!
